eng/tox/tox.ini

```diff
@@ -215,4 +215,17 @@ passenv = *
 setenv =
   DEPENDENCY_TYPE=Minimum
 commands = 
-    {[deptestcommands]commands}
\ No newline at end of file
+    {[deptestcommands]commands}
+
+
+[testenv:apistub]
+skipsdist = false
+skip_install = false
+usedevelop = true
+changedir = {toxinidir}
+deps =
+  astroid 
+commands =
+    # install API stub generator
+    {envbindir}/python -m pip install "git+https://github.com/Azure/azure-sdk-tools.git#subdirectory=packages/python-packages/api-stub-generator&egg=api-stub-generator"
+    apistubgen --pkg-path {toxinidir} --temp-path {envtmpdir} --out-path {envtmpdir}
```

Comment:
What do we intend to do with the generated api stubs?

Reply:
@weshaggard : Currently this is only to find any issue in package in generating API view. In the near future we plan to modify this and upload review to API view automatically if there is a change from last API surface level.

Reply:
OK. Do we have plans to have the api file committed into the repo? Or do we expect to always generate it in the build pipeline?

Reply:
JS is already having an API extracted file in repo for each package. But that's missing for other packages. One of the original plan is to auto generate reviews when new API changes are discovered from previously approved APIs. I think the best option in this case is to have this comparison within API view tool itself since we have all reviews stored in it. One another reason for this plan is due to the fact that .Net SDKs doesn't have any stub file and API view parses the code using internal .Net code parsing libraries.

Reply:
.NET also has an api stub see https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/api/Azure.Core.netstandard2.0.cs but it is generated via another tool currently. I'm hoping that apiview can be used to generate that as well. 

While I like having apiview produce diffs and uploading them somewhere apiview knows about I also think it is a pretty good idea to keep the api stub files in the source repo along side the source code, it could even be the source of truth so that we don't have to make ApiView keep track of the full history including understanding git branching when we are working it multiple branches outside of master.

Reply:
Thanks. Yes, that will be helpful. I will have this added for all languages before we start adding more enhancements in API View. I will create an issue to track it as well.